dts/bindings/serial/renesas,smartbond-uart.yaml

```diff
@@ -0,0 +1,43 @@
+description: Renesas SmartBond(tm) UART
+
+compatible: "renesas,smartbond-uart"
+
+include: [uart-controller.yaml, pinctrl-device.yaml]
+
+properties:
+    reg:
+      required: true
+
+    interrupts:
+      required: true
+
+    peripheral-id:
+      type: int
+      description: |
+        UART peripheral identifier. 0 for UART, 1 for UART2 and 2 for UART3.
```

Comment:
Same issue here -- HAL limitation workaround, or actually programmed into a register?

Reply:
This is used to enable clock for proper instance. I found at least few other drivers doing exactly the same thing so not sure why it's not ok here?

Reply:
Sometimes things get merged that misbehave. There are too many devicetree bindings for the DT maintainers to review, and not all subsystem maintainers catch things like this.